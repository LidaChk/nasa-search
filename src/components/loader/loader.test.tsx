import React from 'react';
import { render, screen } from '@testing-library/react';
import Loader from './loader';

describe('Loader', () => {
  it('renders loader component', () => {
    render(<Loader />);
    const loaderElement = screen.getByLabelText('loading');
    expect(loaderElement).toBeInTheDocument();
  });

  it('has correct structure', () => {
    const { container } = render(<Loader />);
    const loaderDivs = container.firstChild?.childNodes;
    expect(loaderDivs).toHaveLength(4);
  });

  it('has correct CSS classes', () => {
    const { container } = render(<Loader />);
    expect(container.firstChild).toHaveClass('loader');
  });

  it('matches snapshot', () => {
    const { container } = render(<Loader />);
    expect(container.firstChild).toMatchSnapshot();
  });
  afterAll(() => {
    const _dummyComponent: React.FC = () => {
      return <div>dummy</div>;
    };
    render(<_dummyComponent />);
  });
});
