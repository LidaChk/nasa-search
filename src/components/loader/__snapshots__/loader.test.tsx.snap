// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<div
  aria-label="loading"
  class="loader"
>
  <div />
  <div />
  <div />
  <div />
</div>
`;
